### ISSUE TITLE
WordPress.NamingConventions.ValidHookName.NotLowercase does not distinguish string literals from function calls

### ISSUE THREAD

**Issue by mbmjertan** [WordPress/WordPress-Coding-Standards#2055]
## Bug Description

The `WordPress.NamingConventions.ValidHookName.NotLowercase` sniff violation is triggered when the first argument of `apply_filters` is a function call containing an argument that's not entirely lowercase, instead of validating on the return value of the function call:

```
  31 | ERROR   | [ ] Hook names should be lowercase. Expected: Filters::getBlockFilterName('form', 'datatypeselector'), but found: Filters::getBlockFilterName('form', 'dataTypeSelector').
```

## Minimal Code Snippet

The issue happens when running this command:
```bash
 php ./vendor/squizlabs/php_codesniffer/bin/phpcs --standard=WordPress form.php
```

... over a file containing this code:
```php
$formDataTypeSelector = apply_filters(
	Filters::getBlockFilterName('form', 'dataTypeSelector'), 
	Components::checkAttr('formDataTypeSelector', $attributes, $manifest),
	$attributes
);
```

Important to note that the return value of the `Filters::getBlockFilterName('form', 'dataTypeSelector')` method call is `es_forms_block_form_data_type_selector`, which is entirely lowercase.

## Error Code

`WordPress.NamingConventions.ValidHookName.NotLowercase`

## Environment

| Question               | Answer
| ------------------------| -------
| PHP version             | 7.4.29
| PHP_CodeSniffer version | 3.6.2
| WPCS version            | 2.3.0
| WPCS install type       | Composer local install as a dependency of Eightshift Coding Standards

## Tested Against `develop` branch?
- [ ] I have verified the issue still exists in the `develop` branch of WPCS.

**Comment by jrfnl** [comment]
@mbmjertan Thanks for reporting this.

This is a bug in the code, but not in the way you see it. PHPCS/WPCS does not execute the code, so there is absolutely no way for it to know the return value of a function call. And therefore no way to evaluate the return value for being valid based on these rules.
Instead if a function call is detected, the sniff should probably ignore it or flag it for manual inspection.

As a side-question, I wonder if non-text-string hook names should even be allowed.
While this may not be an issue in closed-source. For open source plugins/themes, this obscures the hook name and makes it impossible for people to know what hook to use, especially as documentation generators will also not be able to interpret the function call.

**Comment by mbmjertan** [comment]
> PHPCS/WPCS does not execute the code, so there is absolutely no way for it to know the return value of a function call. 

Ah, this fact is obscured by the amazing reports, which makes it easy to forget that it's neither static code analysis nor actually runs code 😄 Sorry about the confusion there 🙈 

> Instead if a function call is detected, the sniff should probably ignore it or flag it for manual inspection.
I like this approach if WPCS agrees to allow this pattern.

Regarding that discussion, this is actually something I discovered while working on [an Eightshift Forms PR](https://github.com/infinum/eightshift-forms/pull/234), which is an open source project, and we decided to use this approach because of maintainability and hook discoverability. 

This might not be an approach WPCS agrees with (especially for wp.org plugins/themes), though, which is perfectly fine - we can make changes in Eightshift CS in that case.
